File: tests/fixtures/sample.php
<?php

namespace App;

interface Drawable {
    public function draw();
}

trait Logger {
    public function log($message) {
        echo $message;
    }
}

class Shape implements Drawable {
    use Logger;

    private $name;

    public function __construct($name) {
        $this->name = $name;
    }

    public function draw() {
        $this->log("Drawing " . $this->name);
    }
}

function add($a, $b) {
    return $a + $b;
}

File: tests/fixtures/sample.rs
pub struct Point {
    x: i32,
    y: i32,
}

pub enum Color {
    Red,
    Green,
    Blue,
}

pub trait Drawable {
    fn draw(&self);
}

impl Drawable for Point {
    fn draw(&self) {
        println!("Point at ({}, {})", self.x, self.y);
    }
}

pub fn add(a: i32, b: i32) -> i32 {
    a + b
}
